### PR TITLE
Update post meta validation/save

### LINKED ISSUE #14: Free download activation not displaying properly
Getting this each time I just save an existing download without making any changes to it

```
[08-Oct-2020 00:19:15 UTC] PHP Notice:  Undefined index: edd_lfd_activate in D:\MAMPPRO\htdocs\edd\wp-content\plugins\edd-license-free-download\edd-license-free-download.php on line 146
```

### PR DESCRIPTION
**Pull request by robincornett**
Fixes #14 

Proposed changes:
1. Metabox input checks for old "yes" value to see if the plugin is enabled for a given download.
2. Validation for array of products is updated to no longer save empty values.

### CODE REVIEW

**Review by mihaijoldis — APPROVED** [review]
Looks good here

**Review by ashleyfae — CHANGES_REQUESTED** [review]
As a side effect of this, downloads are now always showing the current product selected in the product list.

Here's a screenshot of the HTML to show what I mean:

Screenshot from 2020-10-12 12-14-56

`License Free Download` is already in the list earlier up with its ID, but it's also appended to the bottom with an empty value. This is because this is returning an empty string as a value:

```php
$values   = get_post_meta( $post->ID, '_edd_lfd_products', true );
```

And EDD core's `EDD()->html->product_dropdown()` function, has a section titled:

```php
// This ensures that any selected products are included in the drop down
```

where it adds any selected items into the dropdown, if `$args['selected']` is an array or doesn't equal `false`. In this case it's an empty string, which means it gets added to the product array when it shouldn't.

tl;dr, we need `$values` to be set to `false` if it's empty.

**Review by ashleyfae — APPROVED** [review]
